homework2/task1.py

```diff
@@ -0,0 +1,111 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    """
+
+    Find 10 words with largest amount of unique symbols
+
+    :param file_path: Path to file with text
+    :return: List of 10 most diverse words
+    """
+    with open(file=file_path, mode="r", encoding="unicode-escape",
+              errors="ignore") as fi:
+        unique_words = set()
+        for line in fi:
+            for word in line.strip().split():
+                if word not in unique_words:
+                    unique_words.update([word])
+    most_diverse = dict((x, len(set(x))) for x in unique_words)
+    most_diverse = (sorted(sorted(most_diverse),
```

Comment:
Do not use the same names for different content.
Avoid multiple calls in one line, double `sorted` is ambiguous

Reply:
The problem with double `sorted ` is that most_diverse dict keys are made from unique_words set elements. Set is unordered, so received dict is also unordered. When making a sorted list from it, words with same diversity are being placed in order they were in the dict. That leads to having testing problems, because on every use of `get_longest_diverse_words` function we get different lists (elements are same, but order of words with the same diversity is random). I fixed that with using `sorted` on dict, before making a sorted list from it, so that elements with the same diversity will always be alphabetically ordered. 

I think I'll try replacing unique_words set with most_diverse dict initially, this must result in having dict with the same order (order of word occurrence) every time, so that one `sorted` function could be removed

Reply:
You referred to `most_diverse`, but its unclear to which one, you have many of them.
Cut logic in pieces, if you have to do such tricky thing don't be afraid to left a comment answering question why